src/channel.js

```diff
@@ -2,6 +2,53 @@ import {ascending, descending, rollup, sort} from "d3";
 import {first, isIterable, labelof, map, maybeValue, range, valueof} from "./options.js";
 import {registry} from "./scales/index.js";
 import {maybeReduce} from "./transforms/group.js";
+import {maybeColorChannel, maybeNumberChannel} from "./options.js";
+import {maybeSymbolChannel} from "./symbols.js";
+import {warn} from "./warnings.js";
+
+// An array of known channels, with an associated scale name, and a definition
+// that returns [variable, undefined] if variable, or [undefined, constant] if
+// constant (such as "#eee" for the color channel)
+export const knownChannels = new Map([
```

Comment:
Let’s call this `registry` (or `channelRegistry`) like we do for src/scales/index.js, and be consistent.